scheduler/feasible.go

```diff
@@ -223,11 +235,28 @@ func (h *HostVolumeChecker) hasVolumes(n *structs.Node) bool {
 			}
 
 			if req.Sticky {
-				if slices.Contains(h.hostVolumeIDs, vol.ID) || len(h.hostVolumeIDs) == 0 {
+				// the node is feasible if there are no claims or if there's an
+				// exact match
+				if len(h.claims) == 0 {
 					return true
 				}
 
-				return false
+				claimed := []string{}
+				for _, c := range h.claims {
+					if c.VolumeID == vol.ID {
+						// keep track of claims that we "used"
+						claimed = append(claimed, c.ID)
+						return true
+					}
+				}
+
+				// if we have a match for a volume claim, delete this claim
+				// from the claims list in the feasibility checker. This is
+				// needed for situations when jobs get scaled up and new
+				// allocations need to be placed on the same node.
+				h.claims = slices.DeleteFunc(h.claims, func(c *structs.TaskGroupHostVolumeClaim) bool {
+					return slices.Contains(claimed, c.ID)
+				})
```

Comment:
How do I reach this if I have a match? We return true on line 249 which means we'll never mark that claim as having been used and the next alloc might try to grab it as well.

Reply:
oof. picard_riker_janeway_facepalm.gif

Reply:
fixed in 34f139e